### PR TITLE
Bump Active Model Serializers 

### PR DESCRIPTION
**Pull request by yuenmichelle1**
As we move to Rails 5.1, current version of active_model_serializers (0.10.0rc2) uses a deprecated method in 4.1 (.to_prepare) which blocks app to build in 5.1.

Goal of this PR is to bump up active_model_serializers as high as we can in order for us to build app in 5.1 and see what tests are failing in 5.1 with the dual boot.

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: spec/serializers/event_stream_serializers/classification_serializer_spec.rb [hunk 1]
```diff
@@ -8,7 +10,22 @@
...
+      serialized_data = described_class
+                        .serialize(classification, include: %w[project workflow user subjects])
+                        .as_json
+                        .with_indifferent_access
+
+      sample_kinesis_json = JSON.parse(file_fixture('example_event_stream_serializer_classification.json').read)
+
+      expect(serialized_data.keys).to match_array(%w[classifications linked])
+      expect(serialized_data[:classifications][0].keys).to match_array(sample_kinesis_json['data'].keys)
+      expect(serialized_data[:linked].keys).to match_array(sample_kinesis_json['linked'].keys)
```
Comment (yuenmichelle1) [inline spec/serializers/event_stream_serializers/classification_serializer_spec.rb:28]: We could do even more with this and check the objects within `linked` to view if that matches as expected. (But that also would have been checked in the appropriate associated object's [i.e. projects/workflows/users etc] serializer tests). 

I dont mind adding that as an assertion though if that is what we want as a team. IMO, this feels like enough.

File: app/serializers/event_stream_serializers/classification_serializer.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def self.serialize(classification, options = {})
     belongs_to :user,             serializer: EventStreamSerializers::UserSerializer
     belongs_to :workflow,         serializer: EventStreamSerializers::WorkflowSerializer
     has_many   :subjects,         serializer: EventStreamSerializers::SubjectSerializer
+    type :classifications
```
Comment (camallen) [inline app/serializers/event_stream_serializers/classification_serializer.rb:15]: i think this should go above the relation links for readability, perhaps on line 10/11?

**Review by camallen — COMMENTED** [review]
This looks like great and comprehensive work - i've got a few minor questions.

File: config/environments/test.rb [hunk 3]
```diff
@@ -45,7 +45,7 @@
 
   #hack to fix issue with doorkeeper not loading ControllerHelpers in test env
   # https://github.com/doorkeeper-gem/doorkeeper/issues/375
-  config.to_prepare do
+  ActiveSupport::Reloader.to_prepare do
```
[the remark this answers is not in the document]
Comment (yuenmichelle1) [reply config/environments/test.rb:48]: that is...a very good question. unsure of this, but i bet that we could have gotten away with removing this at 5.0 as well. let me confirm this..

[the remark this answers is not in the document]
Comment (yuenmichelle1) [reply config/environments/test.rb:48]: oh yeah it looks like we can get away with removing this hack now that we're on 5.0. fwiw, (we already mentioned this in slack) my plan is to separate ams upgrade from rails 5.1 upgrade. gonna close this pr and open a new one that solely has ams upgrade changes. 

i'll remove this hack in the Upgrade to 5.1 dual boot PR.

File: lib/serialization/v1_adapter.rb [hunk 4]
```diff
@@ -1,11 +1,12 @@
-require 'active_model/serializer/adapter/json_api/fragment_cache'
+require 'active_model_serializers/adapter/json_api'
```
[the remark this answers is not in the document]
Comment (yuenmichelle1) [reply lib/serialization/v1_adapter.rb]: we sure can. yup.

File: lib/serialization/v1_adapter.rb [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 
 module Serialization
-  class V1Adapter < ActiveModel::Serializer::Adapter
+  class V1Adapter < ActiveModelSerializers::Adapter::Base
     def initialize(serializer, options = {})
       super
       serializer.root = true
-      @hash = { serializer.type => [] }
+      @hash = { serializer._type => [] }
```
[the remark this answers is not in the document]
Comment (yuenmichelle1) [reply lib/serialization/v1_adapter.rb]: ah yeah using `serializer.json_key` should be fine, in order to use it though we had to remove `serializer.root = true` else `serializer.json_key` will return `true` as opposed to `._type` of the serializer.

**Comment by yuenmichelle1** [comment]
Closing in favor of https://github.com/zooniverse/panoptes/pull/3948/files in order to separate concerns of dual booting 5.1 and upgrading AMS

**Pull request closed without merging**